rmf_fleet_adapter/test/phases/MockAdapterFixture.cpp

```diff
@@ -139,31 +135,46 @@ auto MockAdapterFixture::add_robot(
       };
     } ();
 
-  const auto now = rmf_traffic_ros2::convert(adapter->node()->now());
+  const auto now = rmf_traffic_ros2::convert(data->adapter->node()->now());
   const rmf_traffic::agv::Plan::StartSet starts = {{now, 0, 0.0}};
 
   RobotInfo info;
   info.command = std::make_shared<rmf_fleet_adapter_test::MockRobotCommand>(
-    adapter->node(), graph);
+    data->node, data->graph);
 
   std::promise<bool> robot_added;
-  fleet->add_robot(info.command, name, profile, starts,
+  data->fleet->add_robot(info.command, name, profile, starts,
     [&info, &robot_added](rmf_fleet_adapter::agv::RobotUpdateHandlePtr updater)
     {
       const auto& pimpl = agv::RobotUpdateHandle::Implementation::get(*updater);
       info.context = pimpl.context.lock();
       info.command->updater = updater;
       robot_added.set_value(true);
     });
+
   robot_added.get_future().wait();
 
   return info;
 }
 
+//==============================================================================
 MockAdapterFixture::~MockAdapterFixture()
 {
-  adapter->stop();
-  rclcpp::shutdown(_context);
+  std::weak_ptr<rclcpp::Node> weak_node = data->node;
+  data.reset();
```

Comment:
For my understanding:
It seems like with the reset, we're releasing ownership of the `rclcpp::context` shared ptr in `data`.  But then who ends up shutting down the context?

Reply:
If you're asking "Who will call `rclcpp::shutdown(context)`?" there actually isn't a need to do that. The destructor of `rclcpp::Context` will take care of all the shutting down.

As far as I know, the only motive to call `rclcpp::shutdown(context)` would be to make it so that `rclcpp::ok(context)` returns `false`. But we're using lifecycle management to decide when to stop instead of listening to `rclcpp::ok(context)`.